### PR TITLE
fix: use `GITHUB_REF` as documented on docs.github.com/actions

### LINKED ISSUE #157: GITHUB_REF should be not head branch ref but `refs/pull/:prNumber/merge`
Hi 👋 I'm a user of semantic-release, thanks for making our hacking smooth and semantic!

## Bug

According to the [official document](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull_request), GitHub Actions' `GITHUB_REF` should be a reference to PR merge branch like `refs/pull/:prNumber/merge`.

But the test in this repo uses the head branch instead:

https://github.com/semantic-release/env-ci/blob/1f504fde1505be788e51a49e98e12433e3174f05/test/services/github.test.js#L55

If we change this test to make it following the doc, `npm test` will fail.

### Context of this bug report

I want to confirm that current implementation is intentional or not, to fix a problem in `pull_request_target` event support. <del>Currently `semantic-release` triggers a release even from `pull_request_target` event, see [this workflow run](https://github.com/KengoTODA/gradle-semantic-release-plugin/runs/4004544776?check_suite_focus=true#step:9:506) for example.</del> <ins>Current test code uses head branch as `GITHUB_REF`, but it should be a merge branch instead.</ins>

I made a [test to reproduce the problem](https://github.com/KengoTODA/env-ci/commit/4d95c68b5bc388097ec4a8b05e0b919bab56e18e) but not sure how I should fix it.

If current implementation (which doesn't follow the GitHub documentation) is a bug, and we will fix it to return reference to PR merge branch like `refs/pull/:prNumber/merge`; The fix for `pull_request_target` event will be simple:

```patch
diff --git a/services/github.js b/services/github.js
index e951430..cba02c7 100644
--- a/services/github.js
+++ b/services/github.js
@@ -18,13 +18,20 @@ const getPrEvent = ({env}) => {
   return {pr: undefined, branch: undefined};
 };
 
+const getPrNumber = (env) => {
+  const event = env.GITHUB_EVENT_PATH ? require(env.GITHUB_EVENT_PATH) : undefined;
+  return event && event.pull_request ? event.pull_request.number : undefined;
+};
+
 module.exports = {
   detect({env}) {
     return Boolean(env.GITHUB_ACTIONS);
   },
   configuration({env, cwd}) {
     const isPr = env.GITHUB_EVENT_NAME === 'pull_request' || env.GITHUB_EVENT_NAME === 'pull_request_target';
-    const branch = parseBranch(env.GITHUB_REF);
+    const branch = parseBranch(isPr ? `refs/pull/${getPrNumber(env)}/merge` : env.GITHUB_REF);
 
     return {
       name: 'GitHub Actions',
diff --git a/test/services/github.test.js b/test/services/github.test.js
index 1f7215e..86b9e87 100644
--- a/test/services/github.test.js
+++ b/test/services/github.test.js
@@ -92,7 +92,7 @@ test('PR - target', (t) => {
       build: '1246789',
       branch: 'master',
       isPr: true,
-      prBranch: 'pr-branch',
+      prBranch: 'refs/pull/10/merge',
       pr: '10',
       root: '/workspace',
       slug: 'owner/repo',
```

But if not, we probably need to find the head branch ref from `env.GITHUB_EVENT_PATH` and it could be complicated.

### PR DESCRIPTION
**Pull request by KengoTODA**
Current test cases use different `GITHUB_REF` value with GitHub official doc. And if we resolve this mismatch, test cases will fail because it expects that `prBranch` will be the name of head branch.

It is impossible to get the name of head branch from environment variables, so this PR will change both given input and expected output in test cases. If necessary, better to parse a JSON file at `env.GITHUB_EVENT_PATH` to find the name of head branch.

In other words, output of this module will be changed: `prBranch` will point not the name of head branch but the merge branch like `refs/pull/10/merge`. Output will be changed only for `pull_request` event and `pull_request_target` event.

closes #157

### CODE REVIEW

**Review by gr2m — APPROVED** [review]
Great PR, thank you!
